src/components/calcite-icon/utils.ts

```diff
@@ -35,21 +35,24 @@ export async function fetchIcon({
   const size = scaleToPx[scale];
   const id = `${normalizeIconName(icon)}${size}${filled ? "F" : ""}`;
 
-  return new Promise(resolve => {
-    if (requestCache[id]) {
-      return resolve(requestCache[id]);
+  return new Promise(async function(resolve) {
```

Comment:
This top-level Promise isn't needed since the function is already `async`.

Reply:
@jcfranco can you unpack that a little. `async` is completely new to me and I'd appreciate the pointers

Reply:
Sure. Basically when you use the `async` keyword on a function, the browser (or a generator helper if transpiling to ES5) will wrap it in a promise. 🤓